recipes/fmilibrary/all/conanfile.py

```diff
@@ -0,0 +1,148 @@
+from os import path
+import posixpath
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.microsoft import is_msvc_static_runtime, is_msvc
+from conan.tools.files import (
+    apply_conandata_patches, export_conandata_patches, get, copy, rmdir)
+from conan.tools.scm import Version
+from conan.tools.env import Environment, VirtualRunEnv
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+
+required_conan_version = ">=1.53.0"
+
+
+class PackageConan(ConanFile):
+    name = "fmilibrary"
+    description = "C library for importing FMUs"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/modelon-community/fmi-library"
+    topics = ("fmi", "fmi-standard", "fmu")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_fmus": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_fmus": True
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+        self.options["expat"].shared = self.options.shared
+        self.options["minizip"].shared = self.options.shared
+        self.options["zlib"].shared = self.options.shared
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("fmi1/1.0.1")
+        self.requires("fmi2/2.0.4")
+        if self.version >= Version("3.0a2"):
+            self.requires("fmi3/3.0.1")
+        self.requires("expat/2.4.9")
+        self.requires("minizip/1.2.13")
+        self.requires("zlib/[>=1.2.11 <2]")
+        # c99_snprintf -> should be externalised
+
+    def validate(self):
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
```

Comment:
I doubt it will be fixed for v2, but there is an v3 in alpha release where this is more relevant. I have created an issue upstream (https://github.com/modelon-community/fmi-library/issues/93), and will add it as a comment in the recipe for reference.